Adds discussion FAQs (#94)
MetOffice/git-working-practices: learners/discuss.md

## Frequently Asked Questions

1. Does Git default to origin if you run `git fetch` without origin/upstream?

>     From the docs:

Reviewer: Add a link here if it's "from the docs"?

>     From the docs:

Author: ```suggestion
    From the [GitHub documentation](https://git-scm.com/docs/git-fetch):
```

    When no remote is specified, by default the origin remote will be used, unless there’s an upstream branch configured for the current branch.
2. Can you rename the upstream remote?
    Yes see the [spoiler on remotes](../episodes/03-feature-branch.md#some-more-about-remotes).
    You should note that the names origin and upstream are just conventions.

>     You may use any name for the remotes however it may be confusing

Reviewer 2: ```suggestion
    You may use any name for the remotes but it may be confusing
```

    for new collaborators who expect origin/upstream.
3. What do you do if you add the wrong remote link?
    Change the remote url, or remove the remote and add the remote again.
    See the [spoiler on remotes](../episodes/03-feature-branch.md#some-more-about-remotes) for the relevant Git commands.
4. How frequently should you commit changes?
    You learnt in the Introduction to Version Control with Git and GitHub
    lesson that commits should occur frequently and be atomic.
    That is contain one small change at a time.
    Committing regularly every couple of hours and pushing straight after

>     committing helps avoid losing any work.

Reviewer 3: Not sure if you need _..every couple of hours_ (you can do a lot damage in a few hours! and a bit counter to the point about atomic commits)

Author: Fixed with Tim's change :)

> 4. How frequently should you commit changes?
>     You learnt in the Introduction to Version Control with Git and GitHub
>     lesson that commits should occur frequently and be atomic.
>     That is contain one small change at a time.
>     Committing regularly every couple of hours and pushing straight after
>     committing helps avoid losing any work.

Reviewer 2: ```suggestion
4. How often should you commit changes?
    You learnt in the Introduction to Version Control with Git and GitHub
    lesson that commits should occur often and be atomic.
    That is contain one small change at a time.
    Committing often (at most every couple of hours) and pushing straight after
    committing helps avoid losing any work.
```

5. Can you have chains of forks? Or a colleagues fork as a remote?

>     Yes to both! Although the later would be an unusual working practice,

Reviewer 4: ```suggestion
    Yes to both! Although the latter would be an unusual working practice,
```

>     we'd recommend seeking advice before adding a colleagues fork

Reviewer 4: ```suggestion
    we'd recommend seeking advice before adding a colleague's fork
```

    as another remote on your local repository.
    Chains of forks are common. If a GitHub user has read access to your

>     repository they can create a fork.

Reviewer 5: Sorry am late to the party, but why is having a colleague's fork as a remote considered unusual? If anything I'd say this is more likely to happen than chain of forks, which only really happens in open-source-land and is unlikely to be useful for MetOffice organisation repos?
